dockerfiles/Dockerfile.shellcheck

```diff
@@ -1,5 +1,10 @@
-FROM    koalaman/shellcheck-alpine:v0.7.1
-RUN     apk add --no-cache bash make
+# syntax=docker/dockerfile:1.3
+
+FROM koalaman/shellcheck-alpine:v0.7.1 AS base
+RUN apk add --no-cache bash
 WORKDIR /go/src/github.com/docker/cli
-ENV     DISABLE_WARN_OUTSIDE_CONTAINER=1
-COPY    . .
+
+FROM base AS shellcheck
+RUN --mount=type=bind,target=. \
```

Comment:
Should this have `set -eo pipefail` ?

Wondering if we should keep the script, to allow running it locally (non-dockerised)?

Reply:
See https://github.com/docker/cli/pull/3211#discussion_r679068340